src/function/arithmetic/invmod.js

```diff
@@ -0,0 +1,57 @@
+import { factory } from '../../utils/factory.js'
+
+const name = 'invmod'
+const dependencies = ['typed', 'config', 'BigNumber', 'xgcd', 'equal', 'smaller', 'mod', 'floor', 'add']
+
+export const createInvmod = /* #__PURE__ */ factory(name, dependencies, ({ typed, config, BigNumber, xgcd, equal, smaller, mod, floor, add }) => {
+  /**
+   * Calculate the (modular) multiplicative inverse of a modulo b. Solution to the equation `ax ≣ 1 (mod b)`
+   * See https://en.wikipedia.org/wiki/Modular_multiplicative_inverse.
+   *
+   * Syntax:
+   *
+   *    math.invmod(a, b)
+   *
+   * Examples:
+   *
+   *    math.invmod(8, 12)             // returns NaN
+   *    math.invmod(7, 13)             // return 2
+   *    math.invmod(15151, 15122)      // returns 10429
+   *
+   * See also:
+   *
+   *    gcd, xgcd
+   *
+   * @param {number | BigNumber} a  An integer number
+   * @param {number | BigNumber} b  An integer number
+   * @return {number | BigNumber }  Returns an integer number
+   *                              where `invmod(a,b)*a ≣ 1 (mod b)`
+   */
+  return typed(name, {
+    'number, number': (a, b) => {
+      if (a !== floor(a) || b !== floor(b)) throw new Error('Parameters in function invmod must be integer numbers')
+      a = a % b
+      if (b === 0) throw new Error('Divisor must be non zero')
+      let res = xgcd(a, b)
+      if (config.matrix === 'Matrix') res = res._data
```

Comment:
The check `config.matrix === 'Matrix'` is implicit, can you change this to either use `isMatrix(x)` from `/src/utils/is.js` and use `.toArray()` or `.valueOf()` on the Matrix instead of using it's private `._data` property? Or simply call `res.valueOf()` which will return Array for every case (when res is Array, DenseMatrix, or SparseMatrix).